addons/rating/controllers/main.py

```diff
@@ -45,7 +45,15 @@ def action_open_rating(self, token, rate, **kwargs):
         }
         rating.write({'rating': rate, 'consumed': True})
         lang = rating.partner_id.lang or get_lang(request.env).code
-        return request.env['ir.ui.view'].with_context(lang=lang)._render_template('rating.rating_external_page_submit', {
+        # Apply .sudo() to avoid error if public_user does not have this allowed company
+        view_model = request.env['ir.ui.view'].sudo()
```

Comment:
I'm thinking that the rating shouldn't be seen if the user don't have access to that company.

Reply:
I do not fully agree in this case. 
If a user is going to do the rating with the link (with access_token) it should always allow him to do that action and for that reason `.sudo()` is applied (something similar is done in other places, for example sale orders).

Also, do not forget https://github.com/odoo/odoo/issues/163001